components/expenses/ExpensePayeeDetails.js

```diff
@@ -92,7 +93,7 @@ const ExpensePayeeDetails = ({ expense, host, isLoading, borderless, isLoadingLo
             )}
             <Flex flexDirection="column" ml={2} css={{ overflow: 'hidden' }}>
               <Span color="black.900" fontWeight="bold" truncateOverflow>
-                {payee.organization?.name || payee.name}
+                {concatenateDisplayNameWithLegalName(payee.legalName, payee.organization?.name || payee.name)}
```

Comment:
I'm not super familiar with that code, but is it currently working for organizations? Shouldn't we do:
```suggestion
                {concatenateDisplayNameWithLegalName(payee.organization?.legalName || payee.legalName, payee.organization?.name || payee.name)}
```

Reply:
Yes this currently works for organizations without issue. I am not exactly sure why we need this `payee.organization` field. Even for the other places it's used such as a few lines below we use, 

https://github.com/opencollective/opencollective-frontend/blob/87a46eb3231ef975098c0e0b2d9ac0deb50df27c/components/expenses/ExpensePayeeDetails.js#L99-L101

I am uncertain why we have to use this `payee.organization` field. 

@kewitz : Do you have any idea where this `organization` is populated since you seem to have added it sometime back? 🤔 

@Betree : For now I've added your suggestion as I thought its not harmful to make it consistent with adding the `payee.organization?.legalName` but I'll reiterate on this based on @kewitz input. 🤔

Reply:
This is used in the case the user is signing up for the platform through the expense flow. (Non-user receives submit-on-behalf invite)

Reply:
Thanks @kewitz . That explains perfectly. So it seems we need to add the Legal Name to the organization creation form within the expense form. Just so that this PR doesn't get too long I opened a new issue for this at, https://github.com/opencollective/opencollective/issues/4719. Hope that's okay with you @Betree. I'll address that there instead of here. 😄